pkg/services/ngalert/ngalert.go

```diff
@@ -329,16 +329,13 @@ func (ng *AlertNG) init() error {
 func subscribeToFolderChanges(logger log.Logger, bus bus.Bus, dbStore api.RuleStore) {
 	// if folder title is changed, we update all alert rules in that folder to make sure that all peers (in HA mode) will update folder title and
 	// clean up the current state
-	bus.AddEventListener(func(ctx context.Context, e *events.FolderTitleUpdated) error {
-		// do not block the upstream execution
-		go func(evt *events.FolderTitleUpdated) {
```

Comment:
After `InTransaction` has been added to `Update` method, it seems that the context passed to the goroutine updating alert rules might have been already closed by the main thread. It causes `IncreaseVersionForAllRulesInNamespace` to fail with an error:
```
ERROR[11-13|15:50:16] Failed to update alert rules in the folder after its title was changed logger=ngalert error="driver: bad connection" folderUID=c637a39f-0876-4d9f-8162-28fc649b7cb4 folder=folder1_upd
```
As a remedy, it was decided to make the alert rules update operation synchronous.
It also allows to rollback entire nested folders update in case if alert rules update fails.